model/schema/sssom.yaml

```diff
@@ -162,15 +154,14 @@ slots:
         description: (The CURIE of the Uberon term for "anatomical entity".)
       - value: biolink:Gene
         description: (The CURIE of the biolink class for genes.)
-  match_type:
-    description: The kind of match that led to the mapping, e.g. Logical or Lexical.
-    range: match_type_enum
+  mapping_justification:
+    description: A mapping justification is an action (or the written representation of that action) of showing a mapping to be right or reasonable.
+    range: EntityReference
     required: true
-    multivalued: true
     examples:
-      - value: Lexical
+      - value: SSSOMV:Lexical
```

Comment:
Before evaluating if this is a good change for SSSOM, I think the `SSSOMV` vocabulary needs to get fleshed out a bit more? I.e., is there finally a page that is specifically describing all of the terms? Previously, there were some explanations of a few of the possible terms buried in a pile of other irrelevant documentation, which isn't good enough.

This vocabulary also needs a Bioregistry entry ;)

Reply:
Maybe I should remove the example - This vocabulary does not exist yet.. Not even sure its should have the SSSOM namespace, or another: Something like "Vocabulary for Ontology Mapping"? That could be a cool standalone thing to do anyways, and then the prefix wouldn't be SSSOMV. What do you think? A vocabulary in the SSSOM namespace or adjacent (SSSOMV) or something independent in mapping commons org: Vocabulary for Ontology Matching?

Reply:
I too am having trouble evaluating this change. Partly understanding/tracking the purpose of the diverse changes, partly understanding their applications (what does it mean when permission_values are or are not listed?), partly interpreting the `value:` string generally (what does it mean that most do not have full IRIs or CURIEs? are they not in the vocabulary)?  Some of the definitions, n.b. Logical and Lexical, are inadequate, and some of the choices are overlapping.

My observation on that particular point is that the vocabulary name/IRI should be defined as part of the release. Include SSSOMV in the prefix list and make it `https://w3id.org/sssom/schema/voc/` (please note the https protocol, which SSSOM should also be define as). You already have the w3id prefix so this should not be an obstacle to implementation at any point.

Reply:
For the sake of explaining the situation, I created the vocabulary I envision here, with only 3 classes: https://github.com/mapping-commons/ontological-mapping-vocabulary. I propose to build a separate vocabulary, independent of SSSOM, to capture all the activities, match types etc as they are currently prevalent in the mapping community. I know about all the disagreements on terminology, so let us not make that an issue here - this is already in SSSOM - so not currently up for debate.

So this PR should not be about evaluating that vocabulary - I will complete it with what we anyways have right now in SSSOM once the general approach is signed off on. And then anyone can add new terms in there.

The only thing that matters here is this:

- We change the `match_type` field to `mapping_justification` for the reasons outlined in #150 
- We constrain the `mapping_justification` field to be values from a controlled semantic web vocabulary.